src/main/java/ch/njol/skript/SkriptEventHandler.java

```diff
@@ -142,8 +147,11 @@ private static void check(Event event, EventPriority priority) {
 
 			logTriggerStart(trigger);
 			Object timing = SkriptTimings.start(trigger.getDebugLabel());
-
-			trigger.execute(event);
+			if (!Bukkit.isPrimaryThread() && !trigger.getEvent().allowAsynchronous()) {
+				Task.callSync(() -> trigger.execute(event));
```

Comment:
I'm wondering if we should be calling `check` on the main thread too if `allowAsynchronous()` returns false.